packages/kit/src/core/dev/plugin.js

```diff
@@ -197,7 +197,9 @@ export async function create_plugin(config, cwd) {
 					try {
 						if (!req.url || !req.method) throw new Error('Incomplete request');
 
-						const base = `${vite.config.server.https ? 'https' : 'http'}://${req.headers.host}`;
+						const base = `${vite.config.server.https ? 'https' : 'http'}://${
+							req.headers[':authority'] || req.headers.host
```

Comment:
This is what I had suggested in #4362, but I don't know whether it's the most correct option. Perhaps something like `req.httpVersionMajor === 2 ? req.headers[':authority'] : req.headers.host` would be slightly better?

Reply:
```suggestion
							req.httpVersionMajor === 2 ? req.headers[':authority'] : req.headers.host
```

(to make it easy in case that's what we want to do)

Reply:
This will make http3 act as http1

Reply:
What should we be using in HTTP/3? (For the record, this code is just run during local dev, not in the production apps, and Vite doesn't currently use HTTP/3 for anything.)

Reply:
Can you explain what you mean when you say it will make it act as http1? Is there some way that http3 is different than http2 here?

Reply:
It would act like HTTP/1 if we are checking for `req.httpVersionMajor === 2` and we instead get `req.httpVersionMajor === 3`. That said, I have no idea whether there's any native HTTP/3 support in Node, or how we could find ourselves with a `req` object here with `req.httpVersionMajor === 3`.

Unless and until Vite starts sending us HTTP/3 connections here (somehow), I don't think that's something we need to worry about.

Reply:
ah, I didn't understand that @bjon's comment was referring to @Conduitry's suggestion rather than my own code. It makes more sense now :smile: 

anyway, I believe the code as it currently is in this PR might be more correct than the suggestion: https://github.com/nodejs/node/commit/f5c0e282ccf98e17f295c11850649ad19a6fff51

Reply:
:+1: Sounds good to me. I'll mark this conversation resolved.